execution/execution.go

```diff
@@ -165,17 +165,31 @@ func newOperator(expr parser.Expr, storage *engstore.SelectorPool, opts *query.O
 		hints.Func = e.Op.String()
 		hints.Grouping = e.Grouping
 		hints.By = !e.Without
+		var paramOp model.VectorOperator
 
 		next, err := newOperator(e.Expr, storage, opts, hints)
 		if err != nil {
 			return nil, err
 		}
-		a, err := aggregate.NewHashAggregate(model.NewVectorPool(stepsBatch), next, e.Op, e.Param, !e.Without, e.Grouping, stepsBatch)
+
+		if e.Param != nil {
+			paramOp, err = newOperator(e.Param, storage, opts, hints)
```

Comment:
I think we can push this into the `if e.Op == parser.TOPK || e.Op == parser.BOTTOMK` statement. We shouldn't need to validate if `e.Param` is set, because for topk and bottomk the parser will do the validation for us.

Reply:
Yeah.. but i think im preparing to the merge conflict about the quantiles PR.
The `NewHashAggregate` will still have a case where it needs a param.

Reply:
I rebase and i think now makes more sense?